arangod/RocksDBEngine/RocksDBTransactionState.cpp

```diff
@@ -575,7 +521,10 @@ void RocksDBTransactionState::triggerIntermediateCommit() {
   LOG_TOPIC(DEBUG, Logger::ROCKSDB) << "INTERMEDIATE COMMIT!";
 #endif
 
-  internalCommit();
+  Result res = internalCommit();
+  if (res.fail()) {
+    THROW_ARANGO_EXCEPTION(res);
+  }
```

Comment:
How much work would it be to change the function signature to return the `Result` instead of throwing?

Reply:
I would like to change that in a seperate pull request if possible

Reply:
That's fine. There are some other exceptions in here that would need to be removed as well. I just thought I should bring it up given that this is technically introducing a new possible exception.